Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs

```diff
@@ -608,6 +608,19 @@ public void OnMouseEvent (MouseEventArgs a)
     /// <param name="ctrl">If <see langword="true"/> simulates the Ctrl key being pressed.</param>
     public abstract void SendKeys (char keyChar, ConsoleKey key, bool shift, bool alt, bool ctrl);
 
+    /// <summary>
+    ///     Provide handling for the terminal write ANSI escape sequence request.
+    /// </summary>
+    /// <param name="ansiRequest">The <see cref="AnsiEscapeSequenceRequest"/> object.</param>
+    /// <returns>The request response.</returns>
+    public abstract string WriteAnsiRequest (AnsiEscapeSequenceRequest ansiRequest);
```

Comment:
Done here https://github.com/gui-cs/Terminal.Gui/pull/3768/commits/36876b989900f6531e06ba89b3420777f37e0819. But I had to change some internal to public, but at least its work and reduced many duplicated code. Let me know what do you think, because I made as draft and may not look very beautiful.